src/Utility/Hash.php

```diff
@@ -985,7 +985,13 @@ public static function sort(array $data, $path, $dir = 'asc', $type = 'regular')
         $keys = static::extract($result, '{n}.id');
         $values = static::extract($result, '{n}.value');
 
-        $dir = strtolower($dir);
+        if (is_string($dir)) {
+            $dir = strtolower($dir);
+        }
+        if (!in_array($dir, [\SORT_ASC, \SORT_DESC], true)) {
+            $dir = (strtolower($dir) === 'asc') ? \SORT_ASC : \SORT_DESC;
```

Comment:
Why use `strtolower()` again?

Reply:
Sorry, this is just an honest mistake! remove it.